config/src/main/java/org/springframework/security/config/annotation/method/configuration/MethodSecurityConfiguration.java

```diff
@@ -0,0 +1,115 @@
+import org.springframework.security.access.expression.method.ExpressionBasedAuthorizationManagerAfterAdvice;
+import org.springframework.security.access.expression.method.ExpressionBasedAuthorizationManagerBeforeAdvice;
+import org.springframework.security.access.expression.method.MethodSecurityExpressionHandler;
+import org.springframework.security.access.intercept.aopalliance.AuthorizationMethodInterceptor;
+import org.springframework.security.access.intercept.aopalliance.MethodSecurityAuthorizationManagerAdvisor;
+import org.springframework.security.access.method.AuthorizationManagerAfterAdvice;
+import org.springframework.security.access.method.AuthorizationManagerBeforeAdvice;
+import org.springframework.security.access.method.DelegatingAuthorizationManagerAfterAdvice;
+import org.springframework.security.access.method.DelegatingAuthorizationManagerBeforeAdvice;
+import org.springframework.security.access.method.MethodAuthorizationContext;
+
+/**
+ * Base {@link Configuration} for enabling Spring Security Method Security.
+ *
+ * @author Evgeniy Cheban
+ * @see EnableMethodSecurity
+ */
+@Configuration(proxyBeanMethods = false)
+@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+public class MethodSecurityConfiguration implements ImportAware {
+
+	private final MethodSecurityExpressionHandler defaultMethodSecurityExpressionHandler = new DefaultMethodSecurityExpressionHandler();
+
+	private int advisorOrder;
+
+	@Bean
+	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+	MethodSecurityAuthorizationManagerAdvisor methodSecurityAdvisor(AuthorizationMethodInterceptor interceptor,
+			AuthorizationManagerBeforeAdvice<MethodInvocation> beforeAdvice,
+			AuthorizationManagerAfterAdvice<MethodInvocation> afterAdvice) {
+		MethodSecurityAuthorizationManagerAdvisor advisor = new MethodSecurityAuthorizationManagerAdvisor(interceptor,
+				beforeAdvice, afterAdvice);
+		advisor.setOrder(this.advisorOrder);
+		return advisor;
+	}
+
+	@Bean
+	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+	AuthorizationMethodInterceptor authorizationMethodInterceptor(
+			AuthorizationManagerBeforeAdvice<MethodInvocation> beforeAdvice,
+			AuthorizationManagerAfterAdvice<MethodInvocation> afterAdvice) {
+		return new AuthorizationMethodInterceptor(beforeAdvice, afterAdvice);
+	}
+
+	@Bean
+	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+	AuthorizationManagerBeforeAdvice<MethodInvocation> authorizationManagerBeforeAdvice() {
+		List<AuthorizationManagerBeforeAdvice<MethodAuthorizationContext>> beforeAdvices = new ArrayList<>();
+		beforeAdvices.add(new ExpressionBasedAuthorizationManagerBeforeAdvice(methodSecurityExpressionHandler()));
+		beforeAdvices.add(new SecuredAnnotationAuthorizationManagerBeforeAdvice());
+		return new DelegatingAuthorizationManagerBeforeAdvice(beforeAdvices);
+	}
+
+	@Bean
+	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+	AuthorizationManagerAfterAdvice<MethodInvocation> authorizationManagerAfterAdvice() {
```

Comment:
It might not go into this PR, but it would be nice if an application could do:

```java
@Bean 
public AuthorizationManagerBeforeAdvice<MethodInvocation> myBeforeAdvice() {
    // ....
}

@Bean 
public AuthorizationManagerAfterAdvice<MethodInvocation> myAfterAdvice() {
    // ...
}
```

And `MethodSecurityConfiguration` would take that one instead of its own.

This would be the equivalent of supplying a custom `SecurityMetadataSource`

Reply:
Done.